### PR TITLE
Complete Deliverable - Fatima's Pod (Monica Parrillo)

### PR DESCRIPTION
**Pull request by monicaleep**
Felt more comfortable after the group discussions in class yesterday, still not sure I understood the fibonacci one correctly.

### CODE REVIEW

File: monica-deliverable.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+//1. Write a function calculateCube that takes a single number and prints the volume of a cube made from that number.
+ function calculateCube(number){
+   console.log(Math.pow(number,3))
+ }
+ calculateCube(5) //125
+
+//2. Write a function isAVowel that takes a character (i.e. a string of length 1) and returns true if it is a vowel, false otherwise. The vowel could be upper or lower case.
+function isAVowel(char) {
+  var vowels = ['a', 'e', 'i', 'o', 'u' ]
+  if (vowels.includes(char.toLowerCase())) {
```
Comment (ghost) [inline monica-deliverable.js:10]: nice work catching the uppercases!

File: monica-deliverable.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+console.log(getTwoLengths('pepperoni', 'pizza')) //[9,5]
+
+//  4. Write a function getMultipleLengths that accepts a single parameter as an argument: an array of strings. The function should return an array of numbers where each number is the length of the corresponding string.
+const getMultipleLengths = (array) => {
+  const lengths = [];
+  for (let i = 0; i < array.length; i++) {
+    lengths.push(array[i].length)
+  }
```
Comment (ghost) [inline monica-deliverable.js:36]: yes! excellent loop 👍

File: monica-deliverable.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+// 2.Write a function printLongestWord that accepts a single argument, an array of strings. The method should return the longest word in the array. In case of a tie, the method should return the word that appears first in the array.
+const printLongestWord = (arr) => {
+  let longest = arr[0];
+  for (i = 1; i < arr.length; i++) {
+    if (arr[i].length > longest.length) {
+      longest = arr[i];
+    }
+  }
+  return longest;
+}
```
Comment (ghost) [inline monica-deliverable.js:74]: awesome 👍 this is a common pattern in solving these code logic puzzles. assume the first element in an array is the one you want, and use a loop to check if that assumption is true / to change the saved element if it's not true.

File: monica-deliverable.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+// For example, the transmogrified result of 5, 3, and 2 is (5 times 3) to the power of 2 is 225.
+const transmogrify = (a, b, c) => {
+  return Math.pow(a * b, c)
+}
+console.log(transmogrify(5, 3, 2)); //225
+
+
+// 4. Project Euler problem 2
+const fib = (num) => {
+  let a =1,b=2;
```
Comment (ghost) [inline monica-deliverable.js:91]: small detail, but be consistent with your spaces around assignment operators. It's more readable if you add a space before and after the assignment operator

File: monica-deliverable.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  const fibArray = []
+  if (num<1){
+    console.log("Input must be positive");
+  }
+  while(a<num){
+    if(a%2==0){
```
Comment (ghost) [inline monica-deliverable.js:97]: same here with the spaces inside the conditional's boolean expression.

Comment (ghost) [reply monica-deliverable.js:97]: around both the modulus operator and the equality operator

File: monica-deliverable.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+  }
+  let sum = 0;
+ fibArray.forEach(item=>{
+   //add item to the sum only if the item is less than 4million
+   if(item<4000000){
+     sum += item;
+   }
+ });
+ return sum;
+}
```
Comment (ghost) [inline monica-deliverable.js:113]: Nice work getting through this brainy problem! A suggestion to make your code more readable -- use empty lines to break up the 'steps' in your function 'recipe'. For example, your first step is to verify user input. So put the 'ingredients' for that step (aka any variables you need to declare) first, then the loops, conditionals, or other logic that 'does' things. Once you're done writing out that step, leave an empty line so the reader understands that the next lines of code build on the code they've just read. Let me know if you have any questions about this!

File: monica-deliverable.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+sumPositive([1, -4, 7, 12]); //20
+
+//with reduce
+function sumPos(arr){
+  return arr.reduce((acc,val)=>{
+    return val>0 ? val+acc : acc
+  })
+}
```
Comment (ghost) [inline monica-deliverable.js:147]: 💯

**Comment by ghost** [comment]
awesome job! marking complete. feel free to close this PR out

File: monica-deliverable.js (hunk 4 above)
Comment (monicaleep) [reply monica-deliverable.js:91]: Thanks Fatima I will keep an eye out on that

**Comment by monicaleep** [comment]
Thanks for the feedback Fatima! Will keep in mind for upcoming deliverables!

**Pull request closed without merging**